src/geom/elem.C

```diff
@@ -3183,4 +3183,27 @@ void Elem::swap2boundaryedges(unsigned short e1,
     boundary_info->add_edge(this, e1, ids2);
 }
 
+bool
+Elem::is_internal(const unsigned int i) const
+{
+  switch (this->dim())
+  {
+    case 0:
+      return false;
+
+    case 1:
+      return !is_vertex(i);
```

Comment:
Since you use `this->dim`, I'd also use `this->is_vertex`, `this->is_edge`, etc. for consistency. I generally prefer using `this->`, since it makes it clear to anyone reading the code for the first time that you're calling a member function.